src/main/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfig.java

```diff
@@ -317,4 +333,16 @@ public final boolean isMaxRecordPerFileLimited() {
     public final int getMaxRecordsPerFile() {
         return getInt(FILE_MAX_RECORDS);
     }
+
+    public final Template getFilenameTemplate() {
+        String templateStr = getString(FILE_NAME_TEMPLATE_CONFIG);
+        if (templateStr == null) {
+            templateStr = "{{topic}}-{{partition}}-{{start_offset}}";
+            if (getCompressionType() == CompressionType.GZIP) {
+                templateStr += ".gz";
+            }
+        }
+        final String prefix = getString(FILE_NAME_PREFIX_CONFIG);
+        return new Template(prefix + templateStr);
```

Comment:
Technically prefix shouldn't be part of the template string. Maybe `Template` should explicitly support `prefix` parameter that would just add one `StaticTemplatePart` when present?

Reply:
I made the prefix separate again, it's added when the final file name is created as it used to be.